packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -1171,12 +1171,20 @@ export class DatabaseInstanceReadReplica extends DatabaseInstanceNew implements
       throw new Error(`Cannot set 'backupRetention', as engine '${engineDescription(props.sourceDatabaseInstance.engine)}' does not support automatic backups for read replicas`);
     }
 
+    // The read replica instance always uses the same engine as the source instance
+    // but some CF validations require the engine to be explicitely passed when some
+    // properties are specified.
+    const shouldPassEngine = anyDefined(
+      props.domain,
+    );
```

Comment:
How about just

```suggestion
    const shouldPassEngine = props.domain != null;
```

?

Reply:
Yeah ok, this was just in case we wanted to easily add other props later that also require engine to be passed.